File: Frontend/src/models/game.ts
import {Attempt} from "./attempt";

export class Game{
  name: string;
  nrOfTries: number;
  gameId: string | undefined;
  attempts: Attempt[];


  constructor(name: string, nrOfTries: number, gameId: string | undefined, attempts: Attempt[] = []) {
    this.name = name;
    this.nrOfTries = nrOfTries;
    this.gameId = gameId;
    this.attempts = attempts;
  }
}

File: Backend/Mastermind/Models/Game.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Mastermind.Models
{
    public class Game
    {
        public string Name { get; set; }
        public int NrOfTries { get; set; }
        public string GameId { get; set; }
        public List<string> GeneratedPattern { get; set; }
        public List<Attempt> Attempts { get; set; }

        public Game()
        {
            
        }

        public Game(string name, int nrOfTries, string gameId = null, List<string> generatedPattern = null, List<Attempt> attempts = null)
        {
            Name = name;
            NrOfTries = nrOfTries;
            GameId = gameId;
            GeneratedPattern = generatedPattern;
            Attempts = attempts;
        }
    }
}